src/constants/shaders/raymarch.frag

```diff
@@ -279,14 +280,22 @@ void main() {
   float clipNear = 0.0;//-(dot(eyeRay_o.xyz, eyeNorm) + dNear) / dot(eyeRay_d.xyz, eyeNorm);
   float clipFar  = 10000.0;//-(dot(eyeRay_o.xyz,-eyeNorm) + dFar ) / dot(eyeRay_d.xyz,-eyeNorm);
 
-  // Sample the depth texture
-  float depth = texture2D(textureDepth, vUv).r;
+  // Sample the depth/position texture
+  // This is EITHER a view space position or a depth value we'll convert to a view space position
+  vec4 meshViewPos = texture2D(textureDepth, vUv);
+  bool hasDepthValue = usingPositionTexture == 0 && meshViewPos.r < 1.0;
+
   // If there's a depth-contributing mesh at this fragment, we may need to terminate the ray early
-  if (depth < 1.0) {
-    // Get a projection space position from depth and uv, and unproject back to object space
-    vec4 meshProj = vec4(vUv * 2.0 - 1.0, depth * 2.0 - 1.0, 1.0);
-    vec4 meshView = inverseProjMatrix * meshProj;
-    vec4 meshObj = inverseModelViewMatrix * vec4(meshView.xyz / meshView.w, 1.0);
+  if (hasDepthValue || (usingPositionTexture == 1 && meshViewPos.a > 0.0)) {
```

Comment:
I think this could use an extra comment up above to say what r,g,b,a of the position texture are expected to represent.   You might even mention that this is a data format contract with Simularium viewer.